msgcntr/messageforums-component-impl/src/java/org/sakaiproject/component/app/messageforums/MessageForumsMessageManagerImpl.java

```diff
@@ -2091,4 +2092,23 @@ public List findMovedHistoryByMessageId(final Long messageid){
 
 	}
 
+	@Override
+	public void saveDraftRecipients(long msgId, List<DraftRecipient> recipients) {
+		for (DraftRecipient dr : recipients) {
+			getHibernateTemplate().persist(dr);
```

Comment:
Is this always adding a new transient DraftRecipient? if not will need to use merge

Reply:
Confirmed with Paul: when the draft is saved or the message is sent, the old rows are deleted and the new rows are added, so `persist()` is appropriate here. Thanks, @ern!